### PR TITLE
Add Symfony 7 support

### PR DESCRIPTION
**Pull request by ostrolucky**
I was curious what needs to be done so far. Not much so far.

@nicolas-grekas Is that added Container type declaration (instead of ContainerInterface) in `\Symfony\Bridge\Doctrine\ManagerRegistry` legit?

### CODE REVIEW

File: Tests/RegistryTest.php [hunk 1]
```diff
@@ -24,54 +25,57 @@
 
 class RegistryTest extends TestCase
 {
+    /** @var MockObject&Container  */
+    private MockObject $container;
+
+    protected function setUp(): void
+    {
+        $this->container = $this->createMock(Container::class);
```
Comment (stof) [inline Tests/RegistryTest.php]: If you want to reduce the diff, you could simply replace `$container = $this->getMockBuilder(ContainerInterface::class)->getMock();` with `$container = $this->createMock(Container::class);` without also moving it to an instance property (which does not have much benefit as there is no shared setup logic

**Review by nicolas-grekas — APPROVED** [review]
> Is that added Container type declaration (instead of ContainerInterface) in \Symfony\Bridge\Doctrine\ManagerRegistry legit?

yes it is, because we need that kind of container in `resetService()`

File: Tests/RegistryTest.php (hunk 1 above)
Comment (ostrolucky) [reply Tests/RegistryTest.php]: Thanks but I know. My goal wasn't to reduce the diff.

**Review by derrabus — APPROVED** [review]

**Review by stof — APPROVED** [review]

**Comment by nicolas-grekas** [comment]
Could help fix the failing job:
```diff
--- a/.github/workflows/continuous-integration.yml
+++ b/.github/workflows/continuous-integration.yml
@@ -39,7 +39,7 @@ jobs:
           # Test LTS
           - symfony-require: "5.4.*"
             dependencies: "highest"
-            php-version: "8.0"
+            php-version: "8.1"
 
```

**Comment by stof** [comment]
@nicolas-grekas to me, this is a bug in Symfony if installing deps on PHP 8.0 with Symfony 5.4 brings incompatible dependencies between Symfony packages.

**Comment by nicolas-grekas** [comment]
I agree, but I'd wait for someone to be affected to have a closer look :)

**Comment by ostrolucky** [comment]
I think it's just because https://github.com/symfony/translation-contracts/commit/9fca65891b3305a38959d766b3bb13ef35ff0600 was still not tagged

File: DependencyInjection/DoctrineExtension.php [hunk 2]
```diff
@@ -1095,19 +1095,6 @@ private function loadValidatorLoader(string $entityManagerName, ContainerBuilder
...
 
-    /**
-     * @param array<string, mixed> $objectManager
-     * @param string               $cacheName
-     *
-     * @return void
-     *
-     * @psalm-suppress MoreSpecificImplementedParamType
-     */
-    public function loadObjectManagerCacheDriver(array $objectManager, ContainerBuilder $container, $cacheName)
```
Comment (xabbuh) [inline DependencyInjection/DoctrineExtension.php:1106]: What is the reason for this change?

Comment (ostrolucky) [reply DependencyInjection/DoctrineExtension.php:1106]: Same method is defined in parent class, but with changed signature

**Review by xabbuh — APPROVED** [review]

**Review by SenseException — APPROVED** [review]

**Review by derrabus — APPROVED** [review]

**Comment by nicolas-grekas** [comment]
Thank you @ostrolucky
